engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -283,6 +280,15 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
     MetadataEvent(MetadataKey(workflowIdForLogging, None, WorkflowMetadataKeys.Name), MetadataValue(name))
   }
 
+  private def convertJsonToLabels(json: String): Labels = {
+    json.parseJson match {
+      case JsObject(inputs) => Labels(inputs.toVector.collect({
+        case (key, JsString(value)) => Label(key, value)
+      }))
+      case _ => Labels(Vector.empty)
```

Comment:
what is this case supposed to represent?

Reply:
case _ represents all the cases other than JsObjects (including failure) after doing json.parseJson. Since now we validate labels during submission time inside PartialWorkflowSources, it is unlikely that parsing the json string of labels will fail here. But we need to parse again as we only have access to the json string and not actual labels.

Same thing happens in WorkflowStoreSubmitActor.

Reply:
I'm not sure about representing failure. From a quick scan of the JSON parsing code it looks like this would throw a `ParseException` if it's unhappy. If this code is reparsing something that parsed successfully once perhaps it doesn't matter, but maybe the first-pass parser might need to be more robust if it's doing something like this.

Reply:
Yes this code does re-parse the labels. We need to re-parse it as we only have access to json string rather than actual Label. The labels are initially parsed in PartialWorkflowSources, and if they are invalid the submission is rejected. So when the control reaches here (in MaterializeWorkflowDescriptorActor) the labels are already valid.

Reply:
so if it can't be an error, should we be concerned if what comes out of the parse is not a `JsObject`?

Reply:
That case (not being JsObject) is also covered in PartialWorkflowSources. If the labels are not JsObject it will result in error stating 'Invalid workflow labels JSON. Expected a JsObject of "labelKey": "labelValue" values.' and the workflow submission is rejected.